circuit/src/inputs.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{nullifier, unspendable_account};
+
+#[cfg(any(test, feature = "test-helpers"))]
+use crate::storage_proof::test_helpers::{default_proof, ROOT_HASH};
+
+/// Inputs required to commit to the wormhole circuit.
+#[derive(Debug)]
+pub struct CircuitInputs {
+    /// The amount sent in the transaction.
+    pub funding_tx_amount: u64,
+    /// Amount to be withdrawn.
+    pub exit_amount: u64,
+    /// The fee for the transaction.
+    pub fee_amount: u64,
+    /// Raw bytes of the nullifier preimage, used to prevent double spends.
+    pub nullifier_preimage: Vec<u8>,
+    /// Raw bytes of the unspendable account preimage.
+    pub unspendable_account_preimage: Vec<u8>,
+    /// A sequence of key-value nodes representing the storage proof.
+    ///
+    /// Each element is a tuple where the items are the left and right splits of a proof node split
+    /// in half at the expected childs hash index.
+    pub storage_proof: Vec<(Vec<u8>, Vec<u8>)>,
+    /// The root hash of the storage trie.
+    pub root_hash: [u8; 32],
+    /// The address of the account to pay out to.
+    pub exit_account: [u8; 32],
+}
+
+impl Default for CircuitInputs {
+    fn default() -> Self {
+        Self {
+            funding_tx_amount: 0,
+            exit_amount: 0,
+            fee_amount: 0,
+            nullifier_preimage: vec![],
+            unspendable_account_preimage: vec![],
+            storage_proof: vec![],
+            root_hash: [0u8; 32],
+            exit_account: [0u8; 32],
+        }
+    }
+}
```

Comment:
Do we really need a default impl? I think we should be explicit about the fact that `CircuitInputs` needs all fields specified.

Reply:
probably not